### PR TITLE
Add missing elements of SQA templates for contact, xfem and tensor_mechanics

### PR DESCRIPTION
**Pull request by bwspenc**
ref #15968

### CODE REVIEW

**Review by cticenhour — CHANGES_REQUESTED** [review]
The SRS for each of these is missing the `reliability` section. This contains the currently desired code coverage (from `.coverage`) for the module. A sample of this from the EM module is shown below for reference. 

```
!template! item key=reliability
The regression test suite will cover at least 95% of all lines of code within the Electromagnetics
module at all times. Known regressions will be recorded and tracked (see [#maintainability]) to an
independent and satisfactory resolution.
!template-end!
```

File: modules/tensor_mechanics/doc/content/modules/tensor_mechanics/sqa/tensor_mechanics_srs.md [hunk 1]
```diff
@@ -1 +1,24 @@
...
+
+For modeling discrete interfaces between solid elements, this module provides a system similar to that for continuous materials that permits the definition of traction-separattion laws for cohesive behavior. It also has similar support for lower-dimensional elements to represent structural elements such as beams and shells. As for the continuum models, sets of `Material` and `Kernel` classes are defined for these cases.
```
Comment (cticenhour) [inline modules/tensor_mechanics/doc/content/modules/tensor_mechanics/sqa/tensor_mechanics_srs.md:7]:
```suggestion
For modeling discrete interfaces between solid elements, this module provides a system similar to that for continuous materials that permits the definition of traction-separation laws for cohesive behavior. It also has similar support for lower-dimensional elements to represent structural elements such as beams and shells. As for the continuum models, sets of `Material` and `Kernel` classes are defined for these cases.
```

**Comment by bwspenc** [comment]
> The SRS for each of these is missing the `reliability` section. This contains the currently desired code coverage (from `.coverage`) for the module. A sample of this from the EM module is shown below for reference.
> 
> ```
> !template! item key=reliability
> The regression test suite will cover at least 95% of all lines of code within the Electromagnetics
> module at all times. Known regressions will be recorded and tracked (see [#maintainability]) to an
> independent and satisfactory resolution.
> !template-end!
> ```
I was wondering about that. Somehow I thought the module I based this on didn't have that, so I thought maybe we weren't doing that for the modules. However, I just went back and looked, and it was actually there. Anyway, I'll add it!

**Review by cticenhour — APPROVED** [review]
Thanks again @bwspenc!

**Pull request merged**